src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/AndroidCertificatePal.cs

```diff
@@ -36,6 +44,13 @@ public static ICertificatePal FromOtherCert(X509Certificate cert)
 
             AndroidCertificatePal certPal = (AndroidCertificatePal)cert.Pal;
 
+            if (certPal._keyStorePrivateKeyEntry is not null)
+            {
+                var jobjectHandle = new Interop.JObjectLifetime.SafeJObjectHandle();
+                Marshal.InitHandle(jobjectHandle, Interop.JObjectLifetime.NewGlobalReference(certPal.Handle));
```

Comment:
What happens if the certificate gets disposed between line 47 and 50 by another thread?

Since `certPal.Handle` basically the dangerous handle of `_keyStorePrivateKeyEntry` should we do a an add ref / release on that first and use that instead of the `certPal.Handle`?

Reply:
You're right, this isn't threadsafe. I'll add add+release ref.